webextension/privileged/ui/CloudDownloadsView.jsm

```diff
@@ -0,0 +1,760 @@
+    document.getElementById("cloudDownloadTypeIcon").setAttribute("data-provider-icon", providerIcon);
+    document.getElementById("cloudDownloadDetail").setAttribute("value", "Change your download settings any time in");
+    document.getElementById("cloudDownloadPreference").setAttribute("value", "Preferences.");
+
+    const cloudDownloadSave = document.getElementById("cloudDownloadSave");
+    cloudDownloadSave.setAttribute("label", "Save to " + providerDisplayName);
+    cloudDownloadSave.setAttribute("providerKey", providerKey);
+
+    if (providerIcon === CLOUD_PROVIDER_DEFAULT_ICON) {
+      cloudDownloadSave.setAttribute("disabled", true);
+    } else {
+      cloudDownloadSave.removeAttribute("disabled");
+    }
+
+    document.getElementById("cloudDownloadCancel").setAttribute("label", "Not Now");
+    panelCloudNotification = document.getElementById("panelCloudNotification");
+    panelCloudNotification.removeAttribute("hidden");
+    panelCloudNotification.addEventListener("click", this);
+  },
+
+  _addRadioOption(key, providerName, document) {
+    const option = document.createElement("radio");
+    option.id = key;
+    option.type = "radio";
+
+    option.setAttribute("class", "panelCloudNotificationUI-radio");
+    option.setAttribute("label", "Save to " + providerName);
+    option.setAttribute("provider", this._formatProviderName(providerName));
+    return option;
+  },
+
+  _addNotificationMultipleProviders(providersMap, document) {
+    const multiProviderSelect = document.getElementById("multiProviderSelect");
+
+    const providers = [];
+    providersMap.forEach((value, key) => {
+      providers.push({"key": key, "name": value.displayName});
+    });
+
+    // Reset multiProviderSelect by removing any previous
+    // added options
+    while (multiProviderSelect.firstChild) {
+      multiProviderSelect.firstChild.remove();
+    }
+
+    for (const provider of providers) {
+      multiProviderSelect.appendChild(
+        this._addRadioOption(provider.key, provider.name, document));
+    }
+    multiProviderSelect.removeAttribute("hidden");
+  },
+
+  _iconURL(name) {
+    if (this._formatProviderName(name) === "localdownload") {
+      return "moz-icon://" + CloudDownloadsInternal.defaultDownloadDirIconURL + "?size=16";
+    }
+    return new URL(this.stylesURL).origin + "/skin/" + this._formatProviderName(name) + ".svg";
+  },
+
+  _setMenuItemAttributes(menuItem, providerKey, providerDetail, downloadElement) {
+    menuItem.setAttribute("label", providerDetail.displayName);
+    menuItem.setAttribute("providerKey", providerKey);
+    menuItem.setAttribute("class", "menuitem-iconic");
+    menuItem.setAttribute("image", this._iconURL(providerDetail.displayName));
+    return menuItem;
+  },
+
+  _setMoveDownloadMenuPopUpAttributes(menuItem) {
+    menuItem.setAttribute("label", "Move Download To...");
```

Comment:
Nit: use actual ellipsis ( "\u2026") to avoid this looking wrong next to other items that do use the ellipsis char

Reply:
Also, seems you could just set this label once when registering the context menu, instead of doing it every time we use it.

Reply:
Done